addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -610,6 +610,149 @@ QUnit.test('chat window: open / close', async function (assert) {
     );
 });
 
+QUnit.test('chat window (mobile): open channel with no server notification', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.channel'].records.push({
+        id: 20,
+        uuid: 'channel-uuid',
+        state: 'closed'
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerHeight: 640,
+                innerWidth: 360,
+            },
+            device: {
+                isMobile: true,
+            },
+        },
+    });
+    await afterNextRender(() => document.querySelector(`.o_MessagingMenu_toggler`).click());
+    await afterNextRender(() =>
+        document.querySelector(`.o_MessagingMenu_dropdownMenu .o_NotificationList_preview`).click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ChatWindow',
+        "should have a chat window after clicking on thread preview"
+    );
+    let channel = await this.env.services.rpc({
+        model: 'mail.channel',
+        method: 'read',
+        args: [20],
+    }, { shadow: true });
+    assert.strictEqual(
+        channel[0].state,
+        'closed'
+    );
+});
+
+QUnit.test('chat window (mobile): close channel with no server notification', async function (assert) {
+    assert.expect(4);
+
+    this.data['mail.channel'].records.push({
+        id: 20,
+        uuid: 'channel-uuid',
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerHeight: 640,
+                innerWidth: 360,
+            },
+            device: {
+                isMobile: true,
+            },
+        },
+    });
+    await afterNextRender(() => document.querySelector(`.o_MessagingMenu_toggler`).click());
+    await afterNextRender(() =>
+        document.querySelector(`.o_MessagingMenu_dropdownMenu .o_NotificationList_preview`).click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ChatWindow',
+        "should have a chat window after clicking on thread preview"
+    );
+    let channel = await this.env.services.rpc({
+        model: 'mail.channel',
+        method: 'read',
+        args: [20],
+    }, { shadow: true });
+    assert.strictEqual(
+        channel[0].state,
+        'open'
+    );
+
+    // Close chat window
+    await afterNextRender(() => document.querySelector(`.o_ChatWindowHeader_commandClose`).click());
+    assert.containsNone(
+        document.body,
+        '.o_ChatWindow',
+        "should not have a chat window after closing it"
+    );
+    channel = await this.env.services.rpc({
+        model: 'mail.channel',
+        method: 'read',
+        args: [20],
+    }, { shadow: true });
+    assert.strictEqual(
+        channel[0].state,
+        'open'
+    );
+});
+
+QUnit.test('chat window (mobile): chat window doesn\'t pop in front on new message', async function (assert) {
```

Comment:
Use double quote for string if you need to escape single quote inside it.

Reply:
"pop in front on new message" -> "open automatically on receiving a new message"
It's important to use the same "open" keyword as is used elsewhere, and also "receiving" is an important keyboard because a new message could also be "sent" (by the current user)

Reply:
Done